Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -314,6 +314,24 @@ def check_cpuid_instruction(formula)
     "No `cpuid` instruction detected. #{formula} should not use `ENV.runtime_cpu_detection`."
   end
 
+  def check_binary_arches(formula)
+    return unless formula.prefix.directory?
+    # There is no `binary_executable_or_library_files` method for the generic OS
+    return if !OS.mac? && !OS.linux?
+
+    keg = Keg.new(formula.prefix)
+    mismatches = keg.binary_executable_or_library_files.reject do |file|
+      file.arch == Hardware::CPU.arch
```

Comment:
`Hardware::CPU.arch` returns `:x86_64` when running under Rosetta, so this audit should behave as expected on ARM in `/usr/local` too.

Reply:
Though, one corner case worth considering, where I don't know what happens (and don't have the hardware to test...): a user who has an M1 machine with `brew` installed into `/usr/local`, and runs `/usr/local/bin/brew install` without running the process under Rosetta2. I suspect this audit will produce errors for them, but I don't know if `brew` will error out for that case earlier than that.

Reply:
> I don't know if `brew` will error out for that case earlier than that.

it will 🎉